frontend/src/app/pipes/linkify.pipe.ts

```diff
@@ -0,0 +1,24 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+const URL_REGEXP = RegExp(
+  '(https?://' +
+    '(?:[a-z0-9.~_!$&\'()*+,;=-]|%[0-9a-f][0-9a-f])+' + // hostname characters, including hex codes.
+    '(?:/(?:[a-z0-9.~_!$&\'()*+,;=:@-]|%[0-9a-f][0-9a-f])*)*' + // path components
+    '(?:[a-z0-9~_/]))' + // People often follow URLs with punctuation. This will prevent that from getting consumed.
+    '|([a-z~_*+=-]+(?:\\.[a-z0-9~_*+=-]+)*\\.(?:com|net|org))' + // Loose hostname. Much stricter list of accepted characters.
+    '|([a-z0-9+_~.-]+@[a-z~_*+=-]+(?:\\.[a-z0-9~_*+=-]+)*\\.(?:com|net|org))', // Email addresses. Much stricter list of accepted characters.
+  'gi');
+
+function replacer(match: string, p1?: string, p2?: string, p3?:string) {
+  const url = p1 || (p2 && `http://${p2}`) || (p3 && `mailto:${p3}`);
```

Comment:
Hmm.  Should the default assumption be http, or https?  At this point, I have no idea which is actually more common, or safer to assume.

Reply:
usually people will redirect up, but not everyone. so http is safer